### PR TITLE
Add Gemfile and new setting for Github pages

### PR DESCRIPTION
**Pull request by corycaywood**
# Description
These changes are being merged to the `gh-pages` branch.

- Add Gemfile and .gitignore files
  - The Gemfile allows the documentation to be generated and tested locally
- Add parse_block_html setting to kramdown
  - This enables us to use the HTML `<details>` tag with code blocks inside

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by khairul-alam-licon — APPROVED** [review]
👍

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+fabric.properties
+
+*.iml
+modules.xml
+*.ipr
+
+/config/
+
+# Jekyll
+/_site/
```
Comment (minh-rakuten) [inline .gitignore:96]: Just minor thing, better to have blank line end of file. The same for Gemfile.

**Review by minh-rakuten — APPROVED** [review]
👍 👍

File: .gitignore (hunk 1 above)
Comment (corycaywood) [reply .gitignore]: Thanks, I updated this.

Also, I added one more setting for Jekyll: `toc_levels`.

**Pull request merged**